plugins/embedding.cc

```diff
@@ -1334,6 +1336,269 @@ regNearestNeighborsFunction(builtinPackage(),
                             "Return the nearest neighbors of a known row in an embedding dataset",
                             "functions/NearestNeighborsFunction.md.html");
 
+/*****************************************************************************/
+/* Image Wrapper function                                                    */
+/*****************************************************************************/
+
+DEFINE_STRUCTURE_DESCRIPTION(ImageWrapperFunctionConfig);
+
+ImageWrapperFunctionConfigDescription::
+ImageWrapperFunctionConfigDescription()
+{
+    addField("expression",
+             &ImageWrapperFunctionConfig::expression,
+             "");           
+}
+
+ImageWrapperInput::
+ImageWrapperInput()
+{
+}
+
+DEFINE_STRUCTURE_DESCRIPTION(ImageWrapperInput);
+
+ImageWrapperInputDescription::
+ImageWrapperInputDescription()
+{
+    addField("x", &ImageWrapperInput::x,
+             "");
+    addField("y", &ImageWrapperInput::y,
+             "");
+}
+
+DEFINE_STRUCTURE_DESCRIPTION(ImageWrapperOutput);
+
+ImageWrapperOutputDescription::
+ImageWrapperOutputDescription()
+{
+    addField("value", &ImageWrapperOutput::value,
+             "");
+}
+
+ImageWrapperFunction::
+ImageWrapperFunction(MldbServer * owner,
+                         PolyConfig config,
+                         const std::function<bool (const Json::Value &)> & onProgress)
+    : BaseT(owner)
+{
+    functionConfig = config.params.convert<ImageWrapperFunctionConfig>();
+    SqlExpressionMldbScope context(owner);
+    auto boundExpr = functionConfig.expression->bind(context);
+    SqlRowScope scope;
+    embedding = boundExpr(scope, GET_ALL);
+    shape = embedding.getEmbeddingShape();
+}
+
+ImageWrapperFunction::
+~ImageWrapperFunction()
+{
+}
+
+struct ImageWrapperFunctionApplier
+    : public FunctionApplierT<ImageWrapperInput, ImageWrapperOutput> {
+
+    ImageWrapperFunctionApplier(const Function * owner)
+        : FunctionApplierT<ImageWrapperInput, ImageWrapperOutput>(owner)
+    {
+        info = owner->getFunctionInfo();
+    }
+};
+
+ImageWrapperOutput
+ImageWrapperFunction::
+applyT(const ApplierT & applier_, ImageWrapperInput input) const
+{   
+    ImageWrapperOutput output;    
+
+    //We clamp, we do not currently provide interpolation
+    int x = input.x.coerceToInteger().toInt();
+    int y = input.y.coerceToInteger().toInt();
+    x = boost::algorithm::clamp(x, 0, shape[0]-1);
```

Comment:
Instead of clamping, we should return null if the pixel is outside of the image.